src/Our.Umbraco.StackedContent/Models/UnpublishedContent.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Models;
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Core.PropertyEditors;
+using Umbraco.Core.Services;
+using Umbraco.Web.Models;
+
+namespace Our.Umbraco.StackedContent.Models
+{
+    internal class UnpublishedContent : PublishedContentWithKeyBase
+    {
+        private readonly IContent content;
+
+        private readonly Lazy<IEnumerable<IPublishedContent>> children;
+        private readonly Lazy<PublishedContentType> contentType;
+        private readonly Lazy<string> creatorName;
+        private readonly Lazy<IPublishedContent> parent;
+        private readonly Lazy<Dictionary<string, IPublishedProperty>> properties;
+        private readonly Lazy<string> urlName;
+        private readonly Lazy<string> writerName;
+
+        public UnpublishedContent(int id, ServiceContext serviceContext)
+            : this(serviceContext.ContentService.GetById(id), serviceContext)
+        { }
+
+        public UnpublishedContent(IContent content, ServiceContext serviceContext)
+            : base()
+        {
+            Mandate.ParameterNotNull(content, nameof(content));
+            Mandate.ParameterNotNull(serviceContext, nameof(serviceContext));
+
+            var userService = new Lazy<IUserService>(() => serviceContext.UserService);
+
+            this.content = content;
+
+            this.children = new Lazy<IEnumerable<IPublishedContent>>(() => this.content.Children().Select(x => new UnpublishedContent(x, serviceContext)));
```

Comment:
Might want to `.ToList` the child select statement

Reply:
Done! 🤘